src/main/scala/thepackage/routing/UserRoutes.scala

```diff
@@ -0,0 +1,80 @@
+package thepackage.routing
+
+import java.time.Clock
+
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model.headers.{CacheDirectives, `Cache-Control`}
+import akka.http.scaladsl.server.Directives._
+import cats.implicits._
+import play.api.libs.json.Json
+import thepackage.directives.AuthzDirectives
+import thepackage.payloads.requests.{CreateUserRequest, UpdateUserPasswordRequest}
+import thepackage.payloads.responses.{SuccessfulAuthorizationResponse, UserResponse}
+import thepackage.services.{AuthorizationService, UserService}
+import thepackage.util._
+import akka.http.scaladsl.marshalling.PredefinedToResponseMarshallers._
+
+import scala.concurrent.ExecutionContext
+
+class UserRoutes(userService: UserService, authorizationService: AuthorizationService, authzDirectives: AuthzDirectives)
+  (implicit ec: ExecutionContext, clock: Clock) {
+  import authzDirectives._
+
+  val routes =
+    // format: OFF
+    pathPrefix("users") (
+      pathEnd (
+        post (
+          entity(as[CreateUserRequest]) (request =>
+            complete(
+              userService
+                .createUser(request).et
+                .map(Created -> UserResponse.fromUser(_))
+            ))
+        )
+      ) ~
+      requireUserAuthz ( authContext =>
+        pathPrefix("me") (
+          pathEnd (
+            get (
+              complete(
+                userService
+                  .retrieveUser(authContext.userId)
+                  .map(maybeUser => UserResponse.fromUser(maybeUser.get))
+              )
+            )
+          ) ~
+          path("password") (
+            put (
+              entity(as[UpdateUserPasswordRequest]) ( request =>
+                complete(
+                  userService.updateUserPassword(authContext.userId, request.password)
```

Comment:
require re-auth or special auth?

Reply:
I was thinking about it.
Require token to be less than 5 minutes old?